File: Practice/Python/Data Structure/Graph-Based/Adjacency-Matrix.py
# Adjacency Matrix representation in Python
# An Adjacency Matrix is a 2D array of size V x V where V is the number of vertices in a graph. Let the 2D array be adj[][], a slot adj[i][j] = 1 indicates that there is an edge from vertex i to vertex j. Adjacency matrix for undirected graph is always symmetric. Adjacency Matrix is also used to represent weighted graphs. If adj[i][j] = w, then there is an edge from vertex i to vertex j with weight w.
class Graph(object): # Creating a graph
    def __init__(self, size): # Initialize the matrix
        self.adjMatrix = []
        for i in range(size):
            self.adjMatrix.append([0 for i in range(size)])
        self.size = size
    def add_edge(self, v1, v2): # Add edges
        if v1 == v2:
            print("Same vertex %d and %d" % (v1, v2))
        self.adjMatrix[v1][v2] = 1
        self.adjMatrix[v2][v1] = 1
    def remove_edge(self, v1, v2): # Remove edges
        if self.adjMatrix[v1][v2] == 0:
            print("No edge between %d and %d" % (v1, v2))
            return
        self.adjMatrix[v1][v2] = 0
        self.adjMatrix[v2][v1] = 0
    def __len__(self):  # Number of vertices
        return self.size
    def print_matrix(self): # Print the matrix
        for row in self.adjMatrix:
            for val in row:
                print('{:4}'.format(val)),
            print
def main():
    g = Graph(5)
    g.add_edge(0, 1)
    g.add_edge(0, 2)
    g.add_edge(1, 2)
    g.add_edge(2, 0)
    g.add_edge(2, 3)

    g.print_matrix()
if __name__ == '__main__':
    main()

File: Practice/JavaScript/DSA/Data Structure/Graph-Based/Adjacency-Matrix.js
class Graph {
  constructor(numNodes) {
    this.matrix = [];
    for (let i = 0; i < numNodes; i++) {
      this.matrix.push(new Array(numNodes).fill(0));
    }
  }
  addEdge(fromNode, toNode) {
    this.matrix[fromNode][toNode] = 1;
    this.matrix[toNode][fromNode] = 1;
  }
  removeEdge(fromNode, toNode) {
    this.matrix[fromNode][toNode] = 0;
    this.matrix[toNode][fromNode] = 0;
  }
  isEdge(fromNode, toNode) {
    return this.matrix[fromNode][toNode] === 1;
  }
}

const graph = new Graph(5);
graph.addEdge(0, 1);
graph.addEdge(0, 2);
graph.addEdge(1, 3);
graph.addEdge(2, 4);
graph.addEdge(3, 4);
console.log(graph.matrix);
console.log(graph.isEdge(0, 1));
